packages/app/src/common/subscriptions/websocket.ts

```diff
@@ -0,0 +1,290 @@
+         * graphql-ws
+         */
+        new ws.Server({
+          ...cleanObject(optionsObj.wsOptions),
+          noServer: true,
+        }),
+        /**
+         * subscriptions-transport-ws
+         */
+        new ws.Server({
+          ...cleanObject(optionsObj.wsOptions),
+          noServer: true,
+        }),
+      ]
+    : new ws.Server({
+        ...cleanObject(optionsObj.wsOptions),
+        noServer: true,
+      });
+
+  const { buildContext } = optionsObj;
+
+  return function (getEnveloped) {
+    const { schema, execute, subscribe, contextFactory, validate } = getEnveloped();
+
+    async function getContext(contextArgs: BuildSubscriptionContextArgs) {
+      if (buildContext) return contextFactory(Object.assign({}, await buildContext(contextArgs)));
+
+      return contextFactory(contextArgs);
+    }
+
+    let wsTuple: CommonSubscriptionsServerTuple;
+
+    if (enabled === 'new') {
+      assert(!Array.isArray(wsServer));
+      assert(useGraphQLWSServer);
+
+      useGraphQLWSServer(
+        {
+          ...cleanObject(optionsObj.graphQLWS),
+          execute,
+          subscribe,
+          validate,
+          async onSubscribe({ connectionParams, extra: { request, socket } }, { payload: { operationName, query, variables } }) {
+            const { schema, parse } = getEnveloped();
+
+            const args: ExecutionArgs = {
+              schema,
+              operationName: operationName,
+              document: parse(query),
+              variableValues: variables,
+              contextValue: await getContext({
+                connectionParams,
+                request,
+                socket,
+              }),
+            };
+
+            const errors = validate(schema, args.document);
+            if (errors.length) return errors;
+
+            return args;
+          },
+        },
+        wsServer
+      );
+
+      wsTuple = ['new', wsServer];
+    } else if (enabled === 'both') {
+      assert(subscriptionsTransportWs);
+      assert(useGraphQLWSServer);
+      assert(Array.isArray(wsServer));
+
+      useGraphQLWSServer(
+        {
+          ...cleanObject(optionsObj.graphQLWS),
+          execute,
+          subscribe,
+          validate,
+          async onSubscribe({ connectionParams, extra: { request, socket } }, { payload: { operationName, query, variables } }) {
+            const { schema, parse, validate } = getEnveloped();
+
+            const args: ExecutionArgs = {
+              schema,
+              operationName: operationName,
+              document: parse(query),
+              variableValues: variables,
+              contextValue: await getContext({
+                connectionParams,
+                request,
+                socket,
+              }),
+            };
+
+            const errors = validate(schema, args.document);
+            if (errors.length) return errors;
+
+            return args;
+          },
+        },
+        wsServer[0]
+      );
```

Comment:
My review comments from above apply here too.

I'd also recommend refactoring to avoid redundancy by configuring the `useGraphQLWSServer` only once.